src/ILLink.Shared/TrimAnalysis/NullableSystemTypeValue.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using ILLink.Shared.DataFlow;
+using ILLink.Shared.TypeSystemProxy;
+
+namespace ILLink.Shared.TrimAnalysis
+{
+	/// <summary>
+	/// This represents a Nullable<T> where T is a known SystemTypeValue.
```

Comment:
I think I considered that, but Vitek had a reason to leave it as `SingleValue`. I can't remember exactly why, but having it different makes sure that we handle it separately from `SystemTypeValue` instead of accidentally matching `NullableSystemTypeValue` to the `SystemTypeValue` case.

Reply:
I think it's OK to change the underlying type to `SystemTypeValue` if it's always that.
The `NullableSystemTypeValue` must be separate from `SystemTypeValue` - it does act very similarly, but it adds the "this is Nullable flag" to the type.

Reply:
Now I remember we also wanted to keep SystemTypeValue sealed to make type checking faster. I think it might be best to leave it as inheriting from SingleValue.

Reply:
Sorry - what I meant was, can we make the underlyingType a `SystemTypeValue` (not that `NullableSystemTypeValue` should itself be a `SystemTypeValue`)? I think this is more consistent with how the underlying type of a `NullableValueWithDynamicallyAccessedMembers` is also a dataflow value, rather than a `TypeProxy`.

Reply:
Same question for RuntimeTypeHandleForNullableSystemTypeValue.